public/app/plugins/datasource/postgres/datasource.ts

```diff
@@ -71,40 +72,24 @@ export class PostgresDatasource {
     return expandedQueries;
   }
 
-  query(options: any): Observable<DataQueryResponse> {
-    const queries = filter(options.targets, (target) => {
-      return target.hide !== true;
-    }).map((target) => {
-      const queryModel = new PostgresQuery(target, this.templateSrv, options.scopedVars);
-
-      return {
-        refId: target.refId,
-        intervalMs: options.intervalMs,
-        maxDataPoints: options.maxDataPoints,
-        datasourceId: this.id,
-        rawSql: queryModel.render(this.interpolateVariable),
-        format: target.format,
-      };
-    });
-
-    if (queries.length === 0) {
-      return of({ data: [] });
+  filterQuery(query: PostgresQuery): boolean {
+    if (query.hide) {
+      return false;
     }
+    return true;
```

Comment:
Nit, but simpler code:
```suggestion
    return !query.hide;
```